dspace-api/src/main/java/org/dspace/administer/StructBuilder.java

```diff
@@ -571,21 +581,30 @@ public static String getStringValue(Node node) {
         Element[] elements = new Element[communities.getLength()];
 
         for (int i = 0; i < communities.getLength(); i++) {
+            Node tn = communities.item(i);
             Community community;
             Element element = new Element("community");
+            String identifier = tn.getAttributes().getNamedItem("identifier").getNodeValue();
 
             // create the community or sub community
             if (parent != null) {
-                community = communityService.create(parent, context);
+                if (StringUtils.isNotBlank(identifier)) {
+                    community = communityService.create(parent, context, identifier);
+                } else {
+                    community = communityService.create(parent, context);
+                }
             } else {
-                community = communityService.create(null, context);
+                if (StringUtils.isNotBlank(identifier)) {
+                    community = communityService.create(null, context, identifier);
+                } else {
+                    community = communityService.create(null, context);
+                }
```

Comment:
Not your issue, but I noticed that the two branches of the outer 'if' are effectively identical, both before and after your patch.  We could just collapse the outer 'if' to the true path and remove some complexity.